src/sql/platform/connection/common/connectionManagementService.ts

```diff
@@ -543,10 +549,46 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		if (options.showDashboard) {
 			this.showDashboardForConnectionManagementInfo(connectionManagementInfo.connectionProfile);
 		}
+
+		let connectionProfile = connectionManagementInfo.connectionProfile;
 		this._onConnect.fire(<IConnectionParams>{
 			connectionUri: uri,
-			connectionProfile: connectionManagementInfo.connectionProfile
+			connectionProfile: connectionProfile
 		});
+
+		let iconProvider = this._iconProviders.get(connectionManagementInfo.providerId);
+		if (iconProvider) {
+			let serverInfo: azdata.ServerInfo = this.getServerInfo(connectionProfile.id);
+			let iConnectionProfile: azdata.IConnectionProfile = this.toIConnectionProfile(connectionProfile);
```

Comment:
As this code is internal. please use the `interfaces.IConnectionProfile` API definition, and also use the `connectionProfile.toIConnectionProfile` method instead of writing a new one. Otherwise we risk losing connection properties on copy here.

Reply:
fixed.